### PR TITLE
Lgwhl

### PR DESCRIPTION
**Pull request by zakird**
This PR implements --list-of-ips-file which allows scanning a very large number of unrelated IP addresses. This is ready for review.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,8 +17,10 @@ matrix:
         - brew update
         - brew install gengetopt byacc libdnet libunistring
           mongo-c hiredis
-script: cmake -DENABLE_DEVELOPMENT=ON -DWITH_JSON=ON -DWITH_REDIS=ON -DWITH_MONGO=ON
-  . && make
+script: 
+  - cmake -DENABLE_DEVELOPMENT=ON -DWITH_JSON=ON -DWITH_REDIS=ON -DWITH_MONGO=ON
+  - make
+  - python ./scripts/check_manfile.py
```
Comment (dadrian) [inline .travis.yml]: Yay

File: lib/pbm.c [hunk 2]
```diff
@@ -52,3 +59,34 @@ void pbm_set(uint8_t **b, uint32_t v)
...
+	}
+	char line[1000];
+	uint32_t count = 0;
+	while (fgets(line, sizeof(line), fp)) {
+		char *comment = strchr(line, '#');
+		if (comment) {
+			*comment = '\0';
+		}
+		struct in_addr addr;
+		if (inet_aton(line, &addr) != 1) {
```
Comment (dadrian) [inline lib/pbm.c]: Does this handle trailing spaces?

File: src/send.c [hunk 3]
```diff
@@ -327,14 +347,29 @@ int send_run(sock_t st, shard_t *s)
...
 		}
-
+		// number of hosts we actually scanned
 		curr = shard_get_next_ip(s);
+		s->state.sent++;
+		s->state.tried_sent++;
+		if (curr && zconf.list_of_ips_filename) {
+			while (!pbm_check(zsend.list_of_ips_pbm, curr)) {
+				curr = shard_get_next_ip(s);
+				s->state.tried_sent++;
```
Comment (dadrian) [inline src/send.c]: I like this tried_sent thing

File: src/zmap.1 [hunk 4]
```diff
@@ -1,7 +1,7 @@
 .\" generated with Ronn/v0.7.3
 .\" http://github.com/rtomayko/ronn/tree/0.7.3
 .
-.TH "ZMAP" "1" "September 2015" "zmap v2.1.1" "zmap"
+.TH "ZMAP" "1" "April 2016" "" ""
```
Comment (dadrian) [inline src/zmap.1]: You need to manually set the version number at the top here :(

Comment (dadrian) [inline src/zmap.1]: Actually this might be a flag to ronn

File: src/zmap.c [hunk 5]
```diff
@@ -179,20 +180,23 @@ static void start_zmap(void)
...
+	int r;
+	if (!zconf.dryrun) {
+		recv_arg_t *recv_arg = xmalloc(sizeof(recv_arg_t));
+		recv_arg->cpu = zconf.pin_cores[cpu % zconf.pin_cores_len];
+		cpu += 1;
+		r = pthread_create(&trecv, NULL, start_recv, recv_arg);
+		if (r != 0) {
+			log_fatal("zmap", "unable to create recv thread");
+		}
+		for (;;) {
```
Comment (dadrian) [inline src/zmap.c]: This is my favorite code in all of ZMap.

File: src/zmap.c [hunk 6]
```diff
@@ -230,12 +234,12 @@ static void start_zmap(void)
 	}
 	log_debug("zmap", "%d sender threads spawned", zconf.senders);
 
-	monitor_init();
-	mon_start_arg_t *mon_arg = xmalloc(sizeof(mon_start_arg_t));
-	mon_arg->it = it;
-	mon_arg->recv_ready_mutex = &recv_ready_mutex;
-	mon_arg->cpu = zconf.pin_cores[cpu % zconf.pin_cores_len];
-	{
+	if (!zconf.dryrun) {
```
Comment (dadrian) [inline src/zmap.c]: Don't you need to check `--quiet` here?

**Comment by dadrian** [comment]
Looks fine overall, just a couple small comments.

File: src/zmap.c (hunk 6 above)
Comment (zakird) [inline src/zmap.c]: monitor runs even without --quiet. Later on it decides if it should stick around.

**Pull request merged**